crates/cxx-qt-lib/src/gui/qcolor.rs

```diff
@@ -64,11 +280,189 @@ pub struct QColor {
 }
 
 impl QColor {
+    /// Returns the alpha color component of this color.
+    pub fn alpha_f(&self) -> f32 {
+        ffi::qcolor_alpha_f(self)
+    }
+
+    /// Returns the black color component of this color.
+    pub fn black_f(&self) -> f32 {
+        ffi::qcolor_black_f(self)
+    }
+
+    /// Returns the blue color component of this color.
+    pub fn blue_f(&self) -> f32 {
+        ffi::qcolor_blue_f(self)
+    }
+
+    /// Returns a QStringList containing the color names Qt knows about.
+    pub fn color_names() -> ffi::QStringList {
+        ffi::qcolor_color_names()
+    }
+
+    /// Returns the cyan color component of this color.
+    pub fn cyan_f(&self) -> f32 {
+        ffi::qcolor_cyan_f(self)
+    }
+
+    /// Constructs a QColor from the CMYK value c, m, y, k, and the alpha-channel (transparency) value of a.
+    pub fn from_cmyk(c: i32, m: i32, y: i32, k: i32, a: i32) -> Self {
+        ffi::qcolor_init_from_cmyk(c, m, y, k, a)
+    }
+
+    /// Constructs a QColor from the CMYK value c, m, y, k, and the alpha-channel (transparency) value of a.
+    pub fn from_cmyk_f(c: f32, m: f32, y: f32, k: f32, a: f32) -> Self {
+        ffi::qcolor_init_from_cmyk_f(c, m, y, k, a)
+    }
+
+    /// Constructs a QColor from the HSL value h, s, l, and the alpha-channel (transparency) value of a.
+    pub fn from_hsl(h: i32, s: i32, l: i32, a: i32) -> Self {
+        ffi::qcolor_init_from_hsl(h, s, l, a)
+    }
+
+    /// Constructs a QColor from the HSL value h, s, l, and the alpha-channel (transparency) value of a.
+    pub fn from_hsl_f(h: f32, s: f32, l: f32, a: f32) -> Self {
+        ffi::qcolor_init_from_hsl_f(h, s, l, a)
+    }
+
+    /// Constructs a QColor from the HSV value h, s, v, and the alpha-channel (transparency) value of a.
+    pub fn from_hsv(h: i32, s: i32, v: i32, a: i32) -> Self {
+        ffi::qcolor_init_from_hsv(h, s, v, a)
+    }
+
+    /// Constructs a QColor from the HSV value h, s, v, and the alpha-channel (transparency) value of a.
+    pub fn from_hsv_f(h: f32, s: f32, v: f32, a: f32) -> Self {
+        ffi::qcolor_init_from_hsv_f(h, s, v, a)
+    }
+
     /// Constructs a QColor with the RGB value r, g, b, and the alpha-channel (transparency) value of a.
     ///
     /// The color is left invalid if any of the arguments are invalid.
-    pub fn from_rgba(red: i32, green: i32, blue: i32, alpha: i32) -> Self {
-        ffi::qcolor_init_from_rgba(red, green, blue, alpha)
+    pub fn from_rgb(red: i32, green: i32, blue: i32, alpha: i32) -> Self {
```

Comment:
I don't think this should stay named `from_rgba`.
Then add an additional `from_rgb` that uses `255` for alpha.
That's closes to rust-style and to Qt-style I think.
As the `fromRgb` function doesn't require you to provide an alpha component, so it's probably meant to be used without it most of the time anyway.

Reply:
I'm confused what you mean here, i've changed it to `from_rgb`. But do you mean to have both? or do i have it correct ?

Reply:
Basically have: `from_rgba` - the function we have right now, takes `r, g, b, a`
Then also define:
``` rust
pub fn from_rgb(red: i32, green: i32, blue: i32) -> Self {
    Self::from_rgba(red, green, blue, 255)
}
```

Which is kind off how `fromRgb` behaves on the Qt side.
It has a default value of 255 for alpha, so you can call it like this:
`QColor::fromRgb(1, 2, 3)`

Reply:
Hmm but then should we have a `cmyka` and a `hsva` and a `hsla`  ? I ended up copying the Qt naming of dropping the `a` as not required. Then it's unfortunate that rust can't have default values (?) eg we can't do `a = 255` ?

Reply:
Hm, right, I think adding `cmyka`, `hsva`, `hsla`, etc. as well makes sense.

Reasoning:
1. Rust doesn't have optional arguments, the closest thing is this:
    - http://xion.io/post/code/rust-optional-args.html
    - However, it still forces users to write `None` often and is discouraged by Rust maintainers: https://www.reddit.com/r/rust/comments/556c0g/comment/d88m93j/?utm_source=share&utm_medium=web2x&context=3
2. It's a lot more explicit. Having `rgb` take 4 paramter is in itself pretty strange. We're just kind off used to knowing that the alpha value belongs there as well.
3. In Qt most calls to Qt::fromRgb probably only take 3 values, the alpha value being defaulted. So if we want to stick to that same API, `from_rgb` and friends need to accept 3 values only. Providing an alpha value is already the special case (as can be seen in our tests as well).

Reply:
Done adding two methods.